js/gl.js

```diff
@@ -1263,6 +1264,9 @@ var importObject = {
                     wasm_exports.touch(SAPP_EVENTTYPE_TOUCHES_MOVED, touch.identifier, relative_position.x, relative_position.y);
                 }
             });
+            new ResizeObserver(function (_entries) {
```

Comment:
if we use both `onresize` and `ResizeObserver`, wouldn't it generate resize message twice?

Reply:
yes .... especially that `ResizeObserver` is triggered after `window.onresize`, and it has that `force` flag

simply removing `window.onresize` would solve it
but it'll cause the canvas to blink black when resizing the window, probably due to the order things are done in the browser's event loop and the fact that updating the canvas size clears it 

the best solution I found was changing `ResizeObserver` to the following:
```js
new ResizeObserver(() => {
    let dpr = dpi_scale();
    let displayWidth = Math.floor(canvas.clientWidth * dpr);
    let displayHeight = Math.floor(canvas.clientHeight * dpr);
    if (wasm_exports.resize)
        wasm_exports.resize(displayWidth, displayHeight);
}).observe(canvas);
```
and removing the `on_resize` callback from the `resize` function 
(also the `force` flag as it's no longer needed)

Reply:
something like yui-915/miniquad@53b457e4